### PR TITLE
feat: update first unit for more deliberate learning 

### PR DESCRIPTION
**Pull request by mikemaccana**
Would love your feedback, thoughts and suggestions here @jamesrp13. Intention here is to slow down and more deliberately have labs do labs before progressing onto next steps. Also to have students just use web3.js to check balances, then to send sol, then to introduce other on-chain apps interact with the ping counter program. 

 - refactor: make checking balance just use web3.js scripts, rather than having them build a react app. We still cover react in the 'interact with wallets' lesson, we're just focusing more on web3.js itself this early.
 - refactor: split 'writing data' lesson into two lessons with separate labs - one for transferring sol (with a lab where students get devnet sol and transfer it to each other), one for learning how to create instructions for arbitrary apps using the note program.
 - feat: explain atomicness / transactions for students not familiar with these concepts.

### CODE REVIEW

File: content/intro-to-custom-on-chain-programs.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
+---
```
Comment (mikemaccana) [inline content/intro-to-custom-on-chain-programs.md:1]: GitHub's Pull Request UI shows these lessons out of order, this is actually shown after students have made the balance checker and sent each other test SOL.

File: content/intro-to-writing-data.md [hunk 2]
```diff
@@ -66,78 +85,15 @@ The `sendAndConfirmTransaction()` functions takes as parameters
 - a transaction
 - an array of keypairs that will act as signers on the transaction - in this example, we only have the one signer: the sender.
 
-### Instructions
```
Comment (mikemaccana) [inline content/intro-to-writing-data.md:69]: So this has been moved to the next chapter, after students have practiced transferring SOL

File: content/intro-to-reading-data.md [hunk 3]
```diff
@@ -101,90 +103,95 @@ The balance of the account at CenYq6bDRB7p73EjsPEpiYN7uveyPUTdXkDkgUduboaN is 0.
...
-  const addressSubmittedHandler = async (address: string) => {
-    setAddress(address)
-    const key = new web3.PublicKey(address)
-    const connection = new web3.Connection(web3.clusterApiUrl('devnet'));
-    const balance = await connection.getBalance(key);
-    setBalance(balance / web3.LAMPORTS_PER_SOL);
-  }
-  ...
-}
+Save this to a file, and `npx esrun check-balance.ts`. You should something like:
```
Comment (jamesrp13) [inline content/intro-to-reading-data.md:133]: I'm only just now realizing that you're using `esrun`. Haven't used it before. We use ts-node throughout the rest of the course (though it's typically inside a scripts command in the package.json). Is there a reason to use `esrun` over `ts-node`?

Comment (jamesrp13) [reply content/intro-to-reading-data.md:133]: cc @mikemaccana

**Review by jamesrp13 — APPROVED** [review]
Left one comment about `esrun`, would love to hear what you have to say. But I'm all for breaking this up 👍

**Pull request closed without merging**